code/modules/loadout/loadout_items.dm

```diff
@@ -0,0 +1,348 @@
+/proc/init_loadout_categories()
+	var/list/loadout_categories = list()
+	for(var/category_type in subtypesof(/datum/loadout_category))
+		loadout_categories += new category_type()
+
+	sortTim(loadout_categories, /proc/cmp_loadout_categories)
+	return loadout_categories
+
+/proc/cmp_loadout_categories(datum/loadout_category/A, datum/loadout_category/B)
+	var/a_order = A::tab_order
+	var/b_order = B::tab_order
+	if(a_order == b_order)
+		return cmp_text_asc(A, B)
+	return cmp_numeric_asc(a_order, b_order)
+
+/**
+ * # Loadout item datum
+ *
+ * Singleton that holds all the information about each loadout items, and how to equip them.
+ */
+/datum/loadout_item
+	/// The category of the loadout item. Set automatically in New
+	VAR_FINAL/datum/loadout_category/category
+	/// Displayed name of the loadout item.
+	/// Defaults to the item's name if unset.
+	var/name
+	/// Whether this item has greyscale support.
+	/// Only works if the item is compatible with the GAGS system of coloring.
+	/// Set automatically to TRUE for all items that have the flag [IS_PLAYER_COLORABLE_1].
+	/// If you really want it to not be colorable set this to [DONT_GREYSCALE]
+	var/can_be_greyscale = FALSE
+	/// Whether this item can be renamed.
+	/// I recommend you apply this sparingly becuase it certainly can go wrong (or get reset / overridden easily)
+	var/can_be_named = FALSE
+	/// Whether this item can be reskinned.
+	/// Only works if the item has a "unique reskin" list set.
+	var/can_be_reskinned = FALSE
+	/// The abstract parent of this loadout item, to determine which items to not instantiate
+	var/abstract_type = /datum/loadout_item
+	/// The actual item path of the loadout item.
+	var/obj/item/item_path
+	/// Lazylist of additional "information" text to display about this item.
+	var/list/additional_displayed_text
+	/// Icon file (DMI) for the UI to use for preview icons.
+	/// Set automatically if null
+	var/ui_icon
+	/// Icon state for the UI to use for preview icons.
+	/// Set automatically if null
+	var/ui_icon_state
+	/// Reskin options of this item if it can be reskinned.
+	VAR_FINAL/list/cached_reskin_options
+
+/datum/loadout_item/New(category)
+
+	src.category = category
+
+	if(can_be_greyscale == DONT_GREYSCALE)
+		// Explicitly be false if we don't want this to greyscale
+		can_be_greyscale = FALSE
+	else if(item_path::flags_1 & IS_PLAYER_COLORABLE_1)
+		// Otherwise set this automatically to true if it is actually colorable
+		can_be_greyscale = TRUE
+
+	if(isnull(name))
+		name = item_path::name
+
+	if(GLOB.all_loadout_datums[item_path])
+		stack_trace("Loadout datum collision detected! [item_path] is shared between multiple loadout datums.")
+	GLOB.all_loadout_datums[item_path] = src
+
+	if(isnull(ui_icon) && isnull(ui_icon_state))
+		ui_icon = item_path::icon_preview || item_path::icon
+		ui_icon_state = item_path::icon_state_preview || item_path::icon_state
+
+	if(can_be_reskinned)
+		var/obj/item/dummy_item = new item_path()
```

Comment:
Can you check what the total cost of /datum/loadout_item/New in the init_profiler is? I'm suspicious of the performance of this especially as we start to add more items

Reply:
```
/datum/loadout_item/New          0.002        0.002        0.002        0.002           91
/datum/loadout_category/New      0.000        0.002        0.002        0.000            6
```